### ISSUE TITLE
Multiple Machine Rendering

### ISSUE THREAD

**Issue by jcmidia** [Inlife/nexrender#31]
Hi! How can I configure it to render one project in a network?
I tried to create a API server on a machine, and when I started a render node, to look for the ip in another machine, I'm getting this error: connect ECONNREFUSED 127.0.0.1:23234

**Comment by Inlife** [comment]
Hello!

Seems like your render node tries to connect to a api server via ip 127.0.0.1.
Can you somehow show how you are configured the rendernode and project ?

**Comment by jcmidia** [comment]
Hello, thanks for reply, I have solved it :)
The problem is that I was creating a http connection to serve the assets files. But it was configured like this:

`var assets = [
    { "type": "project", "name": "project.aepx",    "src": `http://localhost:${port}/assets/${aepxfile}`},
];`

So, the another machine was looking for http://127.0.0.1:23234/assets/nm.png
Now I changed the src to the IP of the machine running the API and the render node is able to downloading the assets.

Thanks

**Comment by Inlife** [comment]
Wonderful!
Good luck 😃

**Comment by jcmidia** [comment]
Now, How can I use several machine to render a single project?
I created a custom render node like this:

```javascript
var api = require('nexrender').api;
var renderer  = require('nexrender').renderer;

// Configure api connection
api.config({
    host: "localhost",
    port: 3000
});

// connect to api, fetch queued projects, and start rendering
api.get().then(function(results) {
    // iterate, find queued
    for (let project of results) {
        if (project.state === 'queued' || project.state === 'rendering') {
            return startRendering(project);
        }
    }
});

function startRendering(project) {
    var rendered = renderer.render('/Applications/Adobe\ After\ Effects\ CC\ 2015/aerender', project);
    rendered.then(function(project) {
        // apply some custom steps, upload to youtube from there, for example

        // and mark project as finished (updates api state, and triggers remote callbacks)
        project.finish();
    });
}
```

But it seems every machine is rendering on different output folders. What is the best way to configure the renders node to do it?

Thanks so much,

Your project is awesome!

**Comment by Inlife** [comment]
Ok, so in theory what you need to know whether you want to render parts of the video
and then just glue them (in the editor, or via ffmpeg for example)
Or you want to render an image sequence (search for image sequence in AE), and then render all of those images with audio on top

For me first way is easier, however it gives some artifacts on place of "glueing" every 2 videos.

Anyway, for whatever way you choose, the key option to doing this, will be creating multiple projects, using same project file BUT, having `startFrame` and `endFrame` options set up:

Lets say first part (first project):
```js
/* Common project model might look like this: */
{
    template: 'template1.aepx',
    composition: 'composition1',
    settings: {
        outputModule: 'h264',
        outputExt: 'mp4',
        startFrame: 0,
        endFrame: 542
    },
....
```
And second part (other project):
```js
{
    template: 'template1.aepx',
    composition: 'composition1',
    settings: {
        outputModule: 'h264',
        outputExt: 'mp4',
        startFrame: 543,
        endFrame: 1028
    },
....
```

And so on.

This way you will be able to generate multiple different parts of the same project simultaneously, using either image sequence, or partial video rendering.

**Comment by jcmidia** [comment]
Hmm I'm thinking to use image sequence.
Do you think it is possible to configure nexrender to render in a single output folder? Initializing the renderer as Watch folder.
Because my project is about short videos. Today the rendering time is 50 seconds. But the idea is to render under 10s. So, I'm trying to find the best way to do it.

**Comment by Inlife** [comment]
Hm, well

if your renderers are running on the same machine - then its easy
you can 
* start them from the same directory
* setup environment variable RESULTS_DIR pointing to directory where you want it to be rendered
* or doing stuff below:

based on your example, you can add custom stuff inside which will just move all the results to the particular directory manually
```js

function startRendering(project) {
    var rendered = renderer.render('/Applications/Adobe\ After\ Effects\ CC\ 2015/aerender', project);
    rendered.then(function(project) {
        // apply some custom steps, upload to youtube from there, for example
        
        // and mark project as finished (updates api state, and triggers remote callbacks)
        project.finish();
    });
}
```

However if these are different machines, which makes sense, you should use any type of synchronizer for file system. It can be shared folder, it can be dropbox (not really good idea), it can be resilio sync or rsync, or uploading via http onto target machine (o_O), or any other stuff.

What you will be needed to do it just to push rendered files in the shared folder, and after a while it will be synced and will be lying on the target machine. Where you will just need to start new rendering(compositioning) from to turn image sequence into a video with audio

Basically thats it. As you see this functionality (resulted file sync, and multipart rendering) is not (yet) builtin inside nexrender, however it can kinda easily be achieved using any 3rd party tools
